vault/resource_identity_oidc_scope.go

```diff
@@ -0,0 +1,119 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+const identityOIDCScopePathPrefix = "identity/oidc/scope"
+
+func identityOIDCScopeResource() *schema.Resource {
+	return &schema.Resource{
+		Create: identityOIDCScopeCreateUpdate,
+		Update: identityOIDCScopeCreateUpdate,
+		Read:   identityOIDCScopeRead,
+		Delete: identityOIDCScopeDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Description: "The name of the scope. The openid scope name is reserved.",
+				Required:    true,
+			},
+			"template": {
+				Type:        schema.TypeString,
```

Comment:
@benashz working on making this into a map and encoding it to a string when writing it to Vault. What would the conversion look like? Since `EncodeToString` expects a `[]byte`, what's the cleanest way to convert a TypeMap to a `[]byte`? The following doesn't exactly work since you can't convert a map to a byte array

```
if v, ok := d.GetOk(k); ok {
	if k == "template" {
		data[k] = base64.StdEncoding.EncodeToString([]byte(v.(map[string]interface{})))
	} else {
		data[k] = v
	}
}
```

Reply:
It looks like template is `TypeString`, which looks good. The map would be in the calling Terraform code like:

```hcl
template = jsonencode(
   foo = "bar"
)
```

The provider would take that as a string and pass it to vault base64 encoded.

We will probably want to add a `ValidateFunc` to ensure that the the JSON syntax is valid.

Reply:
Fixed in 4848385